From 95599bd85f1c51400d1bbdfd9ce979673b478ffb Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Thu, 18 Nov 2021 18:15:58 +0100
Subject: [PATCH] Revert "usb: dwc2: hcd: fix power down exiting by system
 resume"

At runtime, in partial power down mode, the bus resume does more than
exiting the power saving mode, it also re-initialize the hcd driver.
This makes bus reset to be generated upon each runtime bus_resume
get called. It is also no longer necessary. The original bus resume has
been fixed already.
This reverts commit 950158c37fbf965048da59b09f6750a06470950c.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: Ic69ac53e444dfd77f632fed42cfa1cd88ee431cd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/512988
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 drivers/usb/dwc2/hcd.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/drivers/usb/dwc2/hcd.c b/drivers/usb/dwc2/hcd.c
index 5900ea68077e..2469c492d566 100644
--- a/drivers/usb/dwc2/hcd.c
+++ b/drivers/usb/dwc2/hcd.c
@@ -4433,21 +4433,21 @@ static int _dwc2_hcd_resume(struct usb_hcd *hcd)
 	if (hsotg->lx_state != DWC2_L2)
 		goto unlock;
 
-	switch (hsotg->params.power_down) {
-	case DWC2_POWER_DOWN_PARAM_PARTIAL:
-		hprt0 = dwc2_read_hprt0(hsotg);
+	hprt0 = dwc2_read_hprt0(hsotg);
 
-		/*
-		 * Added port connection status checking which prevents exiting from
-		 * Partial Power Down mode from _dwc2_hcd_resume() if not in Partial
-		 * Power Down mode.
-		 */
-		if (hprt0 & HPRT0_CONNSTS) {
-			set_bit(HCD_FLAG_HW_ACCESSIBLE, &hcd->flags);
-			hsotg->lx_state = DWC2_L0;
-			goto unlock;
-		}
+	/*
+	 * Added port connection status checking which prevents exiting from
+	 * Partial Power Down mode from _dwc2_hcd_resume() if not in Partial
+	 * Power Down mode.
+	 */
+	if (hprt0 & HPRT0_CONNSTS) {
+		set_bit(HCD_FLAG_HW_ACCESSIBLE, &hcd->flags);
+		hsotg->lx_state = DWC2_L0;
+		goto unlock;
+	}
 
+	switch (hsotg->params.power_down) {
+	case DWC2_POWER_DOWN_PARAM_PARTIAL:
 		ret = dwc2_exit_partial_power_down(hsotg, 0, true);
 		if (ret)
 			dev_err(hsotg->dev,
-- 
2.53.0

